packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1263,6 +1263,10 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
 
   private readonly domain: CfnDomain;
 
+  private accessPolicy?: ElasticsearchAccessPolicy
```

Comment:
instead of setting class property for `ElasticsearchAccessPolicy`, save `accessPolicyStatements` instead. then, use [`Lazy`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Lazy.html) in the construction of `ElasticsearchAccessPolicy` to allow both setting access policies in the `Domain` constructor and `addAccessPolicies`